pkg/config/config.go

```diff
@@ -91,26 +90,24 @@ type DeploymentGroup struct {
 	Kind             ModuleKind
 }
 
-func (g DeploymentGroup) getModuleByID(modID string) (Module, error) {
-	idx := slices.IndexFunc(g.Modules, func(m Module) bool { return m.ID == modID })
-	if idx == -1 {
-		return Module{}, fmt.Errorf("%s: %s", errorMessages["invalidMod"], modID)
-	}
-	return g.Modules[idx], nil
-}
-
-func (dc DeploymentConfig) getGroupByID(groupID string) (DeploymentGroup, error) {
-	groupIndex := slices.IndexFunc(dc.Config.DeploymentGroups, func(d DeploymentGroup) bool { return d.Name == groupID })
-	if groupIndex == -1 {
-		return DeploymentGroup{}, fmt.Errorf("%s: %s", errorMessages["groupNotFound"], groupID)
+// Module return the module with the given ID
+func (bp *Blueprint) Module(id string) (*Module, error) {
+	var found *Module
```

Comment:
Elsewhere, we typically use `found` as a bool indicating success of a search. I'd rather this remain a name like `mod` or similar to indicate its type.

Reply:
I think `GetModule()` would be a better func name here. The current name suggests a constructor.

Reply:
> Elsewhere, we typically use found ...

done

> I think GetModule() would be a better func name here. 

https://go.dev/doc/effective_go#Getters